### PR TITLE
type: fix type errors of React.Key

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

Introduced by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66723

https://github.com/react-component/tree/pull/759
https://github.com/react-component/field-form/pull/617

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix `@types/react@18.2.22` React.Key type errors.  |
| 🇨🇳 Chinese |   修复 `@types/react@18.2.22` React.Key 定义更新引发的问题。   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f40c7fd</samp>

The changelog files for version 5.9.2 were updated to fix some discrepancies and errors. The release date was corrected in `CHANGELOG.en-US.md` and some missing bug fixes were added from `CHANGELOG.zh-CN.md`.

### <samp>🤖 Generated by Copilot at f40c7fd</samp>

* Update the release date of version 5.9.2 to 2023-09-19 in both English and Chinese changelogs ([link](https://github.com/ant-design/ant-design/pull/44938/files?diff=unified&w=0#diff-e961ad288c43a2feb2d56b696e10c0b670d652dbb17c76c9aa0452d45157977cL21-R21), [link](https://github.com/ant-design/ant-design/pull/44938/files?diff=unified&w=0#diff-1224d5ca5233fdeba67ba2e2dc6fa3e615905253f35dc132c4b4b86f51f9eab4L21-R25))
* Add four missing bug fixes to the English changelog, related to Table, Select, Modal and notification components ([link](https://github.com/ant-design/ant-design/pull/44938/files?diff=unified&w=0#diff-1224d5ca5233fdeba67ba2e2dc6fa3e615905253f35dc132c4b4b86f51f9eab4L21-R25))

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ tag: vVERSION
 
 ## 5.9.2
 
-`2023-09-18`
+`2023-09-19`
```
[the remark this answers is not in the document]
Comment (afc163) [reply CHANGELOG.en-US.md:21]: 图片

本地 npm run tsc 有报错

**Review by zombieJ — APPROVED** [review]

**Review by kiner-tang — APPROVED** [review]

**Review by RedJue — APPROVED** [review]

**Comment by kiner-tang** [comment]
咋多了这么多报错
image

**Comment by afc163** [comment]
@types/react 又更新了。

**Comment by afc163** [comment]
- [ ] https://github.com/react-component/tree/pull/759
- [ ] https://github.com/react-component/field-form/pull/617

File: CHANGELOG.en-US.md [hunk 2]
```diff
@@ -18,12 +18,14 @@ tag: vVERSION
...
 
 - 🐞 Fix Table selection column not align in center when `size` is `small`. [#44922](https://github.com/ant-design/ant-design/pull/44922)
 - 🐞 Fix Select style problem when `label` contains `div` element. [#44927](https://github.com/ant-design/ant-design/pull/44927)
 - 🐞 Fix Modal broken style of buttons when custom `footer`. [#44929](https://github.com/ant-design/ant-design/pull/44929) [@Wxh16144](https://github.com/Wxh16144)
 - 🐞 Fix notification wrong pop-up animation when `placement` is `bottom`. [#44918](https://github.com/ant-design/ant-design/pull/44918) [@linxianxi](https://github.com/linxianxi)
+- TypeScript
+  - 🤖 Fix React.Key type errors introduced by `@types/react@18.2.22`. [#44938](https://github.com/ant-design/ant-design/pull/44938)
```
Comment (afc163) [inline CHANGELOG.en-US.md:28]:
```suggestion
  - 🤖 MISC: Fix React.Key type errors introduced by `@types/react@18.2.22`. [#44938](https://github.com/ant-design/ant-design/pull/44938)
```

File: CHANGELOG.zh-CN.md [hunk 3]
```diff
@@ -18,12 +18,14 @@ tag: vVERSION
...
-`2023-09-18`
+`2023-09-19`
 
 - 🐞 修复 Table `small` 尺寸时选择列没有居中对齐的问题。[#44922](https://github.com/ant-design/ant-design/pull/44922)
 - 🐞 修复 Select 当 `label` 内使用了 `div` 块级元素时的样式问题。[#44927](https://github.com/ant-design/ant-design/pull/44927)
-- 🐞 修复 Modal 自定义 `footer` 时按钮内容丢失的问题。[#44929](https://github.com/ant-design/ant-design/pull/44929) [](https://github.com/Wxh16144)
+- 🐞 修复 Modal 自定义 `footer` 时按钮内容丢失的问题。[#44929](https://github.com/ant-design/ant-design/pull/44929) [@Wxh16144](https://github.com/Wxh16144)
 - 🐞 修复 notification 底部弹出动画的问题。[#44918](https://github.com/ant-design/ant-design/pull/44918) [@linxianxi](https://github.com/linxianxi)
+- TypeScript
+  - 🤖 修复 `@types/react@18.2.22` React.Key 定义更新引发的问题。[#44938](https://github.com/ant-design/ant-design/pull/44938)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md:28]:
```suggestion
  - 🤖 杂项：修复 `@types/react@18.2.22` React.Key 定义更新引发的问题。[#44938](https://github.com/ant-design/ant-design/pull/44938)
```

**Review by crazyair — APPROVED** [review]

File: CHANGELOG.en-US.md [hunk 4]
```diff
@@ -24,7 +24,12 @@ tag: vVERSION
 - 🐞 Fix Select style problem when `label` contains `div` element. [#44927](https://github.com/ant-design/ant-design/pull/44927)
 - 🐞 Fix Modal broken style of buttons when custom `footer`. [#44929](https://github.com/ant-design/ant-design/pull/44929) [@Wxh16144](https://github.com/Wxh16144)
 - 🐞 Fix notification wrong pop-up animation when `placement` is `bottom`. [#44918](https://github.com/ant-design/ant-design/pull/44918) [@linxianxi](https://github.com/linxianxi)
+<<<<<<< HEAD
 - 🐞 Fix missing inherited feedbackIcon in Form.Item with `noStyle`. [#44937](https://github.com/ant-design/ant-design/pull/44937)
+=======
+- TypeScript
+  - 🤖 MISC: Fix React.Key type errors introduced by `@types/react@18.2.22`. [#44938](https://github.com/ant-design/ant-design/pull/44938)
+>>>>>>> 481ed2969 (Apply suggestions from code review)
```
Comment (yoyo837) [inline CHANGELOG.en-US.md:32]: 冲突未手工处理

Comment (afc163) [inline CHANGELOG.en-US.md:32]:
```suggestion
- 🐞 Fix missing inherited feedbackIcon in Form.Item with `noStyle`. [#44937](https://github.com/ant-design/ant-design/pull/44937)
```

**Comment by bombillazo** [comment]
Seems like there is a draft to revert the change? https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66735

**Comment by zombieJ** [comment]
> Seems like there is a draft to revert the change? [DefinitelyTyped/DefinitelyTyped#66735](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66735)

`string` type only maybe not a good choice. Since it's very common to write like this:

```tsx
<div key={index} />
```

`string` type will beak the usage.

**Pull request merged**